systemdspawner/systemd.py

```diff
@@ -76,73 +76,73 @@ async def start_transient_service(
     slice=None,
 ):
     """
-    Start a systemd transient service with given parameters
+    Start a systemd transient service using systemd-run with given command-line
+    options and systemd unit directives (properties).
+
+    systemd-run ref:             https://www.freedesktop.org/software/systemd/man/systemd-run.html
+    systemd unit directives ref: https://www.freedesktop.org/software/systemd/man/systemd.directives.html
     """
 
     run_cmd = [
         "systemd-run",
-        "--unit",
-        unit_name,
+        f"--unit={unit_name}",
+        f"--working-directory={working_dir}",
```

Comment:
Is passing arguments between spaces not supported? I find it safer for things like escaping/quoting paths to pass the string as a whole argument whenever that's an option, rather than injecting them into a single string with `=`. So unless there's a strong reason, I'd rather go with `'--unit', unit_name, '--working-directory', working_dir` if systemd allows it.

Reply:
Resolved by 70df336.

---

I think either works here and I just refactored this because I find this pattern easier to read and understand in general, mainly because it makes it unambiguous that a value for a flag is just that, and not a positional argument separate from the flag.

> I find it safer for things like escaping/quoting paths to pass the string as a whole argument whenever that's an option, rather than injecting them into a single string with `=`.

I've proposed f`"--flag={value}"` in all situations where its been supported in code I've written, so I'm looking for a bit further motivation for perhaps systematically adjusting back to a `"--flag", value` a pattern. Do you have an example situation when `f"--flag={value}"` could be an issue when `"--flag", value` wouldn't be?

Reply:
Yeah, I get that and it's one of the few things I miss about the pre-autoformatting days. In manual formatting, I prefer to do:

```
"--flag", option,
"--other-flag", other_option,
"--no-arg-flag",
"--more",
```

but that's not available anymore.

> Do you have an example situation when f"--flag={value}" could be an issue when "--flag", value wouldn't be?

I honestly don't, and it may in fact always be fine. It's just that I know that the application is responsible for parsing the string when there's an `=`, and it's not when it's a separate argument from the system, so there's less opportunity to get it wrong. Having written some of that code myself in traitlets, I would always prefer to pass separate values as separate values rather than constructing strings from separate values only to have them parsed back into the original strings when it's supported, but it is not the biggest deal in the world.

Reply:
Thank you @minrk!

Reply:
> but that's not available anymore.

It needs some manual work, but for my scripts I just disable the formatting for command line lists. It takes a way a bit of the utility of black, but it does make it clearer what's a semantic unit.